dpnp/dparray.pyx

```diff
@@ -531,6 +549,14 @@ cdef class dparray:
         if order is not b'C':
             utils.checker_throw_value_error("dparray::reshape", "order", order, b'C')
 
+        if dn:
```

Comment:
This additions apply serious changes on our API and may lead big difference from the original `numpy`. Please, move this changes into other PR and add request review also from O. Pavlyk.